modules/nf-core/mmseqs/search/main.nf

```diff
@@ -0,0 +1,66 @@
+
+process MMSEQS_SEARCH {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda "bioconda::mmseqs2=14.7e284"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mmseqs2:14.7e284--pl5321h6a68c12_2':
+        'biocontainers/mmseqs2:14.7e284--pl5321h6a68c12_2' }"
+
+    input:
+    tuple val(meta), path(db_query)
+    tuple val(meta2), path(db_target)
+
+    output:
+    tuple val(meta), path("${prefix}"), emit: db_search
+    path "versions.yml"               , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+
+    if ("$db_query" == "${prefix}" || "$db_target" == "${prefix}"  ) error "Input and output names of databases are the same, set prefix in module configuration to disambiguate!"
+    """
+    mkdir -p ${prefix}
+
+    # Extract basename of DB
+    DB_QUERY_PATH_NAME=\$(find -L "$db_query/" -maxdepth 1 -name "*.dbtype" | sed -e 'N;s/^\\(.*\\).*\\n\\1.*\$/\\1\\n\\1/;D' )
```

Comment:
This will work in the case that the DBs have been generated by the `MMSEQS_CREATEDB` module but could it be that if the DB has been generated otherwise, this line is not general enough?

Reply:
I'm not certain about that, i think it's quite general. It might not work if the same directory contains multiple mmseqs databases. The `sed` command will identify the longest common substring which corresponds to the sequence database.  
What do u suggest to change to make it more general? Looking for the `.dbtype` is also the the strategy of  mmseqs (see all the `if notExists` statements https://github.com/soedinglab/MMseqs2/blob/f5f780acd64482cd59b46eae0a107f763cd17b4d/data/workflow/nucleotide_clustering.sh#L53). 
Also for the module `mmseqs/cluster` I used the nf-core test database of mmseqs and it is still able to identify the sequence database as it is the one with the shortest string there. 
Maybe the best solution here is to add a `args2` & `args3` to define a search pattern?

Reply:
I think that if looking for the `.dbtype` is the strategy of mmseqs it should be fine. Actually, I was referring more to the fact of calling the directory and the db files the same way which could not be always the case is it?